pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,262 @@
+          or Series.
+        - Output from rows has the same index columns as the original grouped DataFrame
+          or Series. (nth behaves like an aggregator and removes the non-grouped
+          indexes.)
+        - GroupBy.rows can  define a slice relative to the last row of each group.
+        - GroupBy.rows is faster than nth.
+        - GroupBy.rows does not handle dropna.
+
+        An important use case for GroupBy.rows is a multi-indexed DataFrame with a
+        large primary index (Date, say) and a secondary index sorted to a different
+        order for each Date.
+        To reduce the DataFrame to a middle slice of each Date:
+
+            ``df.groupby("Date").rows[5:-5]``
+
+        This returns a subset of df containing just the middle rows for each Date
+        and with its original order and indexing preserved.
+
+        To reduce the DataFrame to the remaining rows:
+
+            ``df.groupby("Date").rows[:5, -5:]``
+
+        Returns
+        -------
+        Series
+            The filtered subset of the original grouped Series.
+        DataFrame
+            The filtered subset of the original grouped DataFrame.
+
+        See Also
+        --------
+        DataFrame.iloc : Purely integer-location based indexing for selection by
+            position.
+        GroupBy.head : Return first n rows of each group.
+        GroupBy.tail : Return last n rows of each group.
+        GroupBy.nth : Take the nth row from each group if n is an int, or a
+            subset of rows, if n is a list of ints.
+
+        Examples
+        --------
+            >>> df = pd.DataFrame([["a", 1], ["a", 2], ["a", 3], ["b", 4], ["b", 5]],
+            ...                   columns=["A", "B"])
+            >>> df.groupby("A", as_index=False).rows[1:2]
+               A  B
+            1  a  2
+            4  b  5
+
+            >>> df.groupby("A", as_index=False).rows[1, -1]
+               A  B
+            1  a  2
+            2  a  3
+            4  b  5
+        """
+        return _rowsGroupByIndexer(self)
+
+
+@doc(GroupByIndexingMixin.rows)
+class _rowsGroupByIndexer:
+    def __init__(self, grouped):
+        self.grouped = grouped
+
+    def __getitem__(self, arg):
+        with groupby.group_selection_context(self.grouped):
+            self._cached_ascending_count = None
+            self._cached_descending_count = None
+
+            if isinstance(arg, tuple):
+                if all(isinstance(i, int) for i in arg):
+                    mask = self._handle_list(arg)
+
+                else:
+                    mask = self._handle_tuple(arg)
+
+            elif isinstance(arg, slice):
+                mask = self._handle_slice(arg)
+
+            elif isinstance(arg, int):
+                mask = self._handle_int(arg)
+
+            elif isinstance(arg, list):
+                mask = self._handle_list(arg)
+
+            else:
+                try:
```

Comment:
remove this path (we simply don't support other things except explicit ones)

Reply:
Done